repositories.bzl

```diff
@@ -0,0 +1,559 @@
+        "include",
+    ],
+)"""
+
+    native.new_git_repository(
+        name = "spdlog_git",
+        remote = "https://github.com/gabime/spdlog.git",
+        commit = "1f1f6a5f3b424203a429e9cb78e6548037adefa8",
+        build_file_content = BUILD,
+    )
+
+    if bind:
+        native.bind(
+            name = "spdlog",
+            actual = "@spdlog_git//:spdlog",
+        )
+
+def tclap_repositories(bind=True):
+    BUILD = """
+cc_library(
+    name = "tclap",
+    hdrs = [
+        "include/tclap/Arg.h",
+        "include/tclap/ArgException.h",
+        "include/tclap/ArgTraits.h",
+        "include/tclap/CmdLine.h",
+        "include/tclap/CmdLineInterface.h",
+        "include/tclap/CmdLineOutput.h",
+        "include/tclap/Constraint.h",
+        "include/tclap/DocBookOutput.h",
+        "include/tclap/HelpVisitor.h",
+        "include/tclap/IgnoreRestVisitor.h",
+        "include/tclap/MultiArg.h",
+        "include/tclap/MultiSwitchArg.h",
+        "include/tclap/OptionalUnlabeledTracker.h",
+        "include/tclap/StandardTraits.h",
+        "include/tclap/StdOutput.h",
+        "include/tclap/SwitchArg.h",
+        "include/tclap/UnlabeledMultiArg.h",
+        "include/tclap/UnlabeledValueArg.h",
+        "include/tclap/ValueArg.h",
+        "include/tclap/ValuesConstraint.h",
+        "include/tclap/VersionVisitor.h",
+        "include/tclap/Visitor.h",
+        "include/tclap/XorHandler.h",
+        "include/tclap/ZshCompletionOutput.h",
+    ],
+    defines = [
+        "HAVE_LONG_LONG=1",
+        "HAVE_SSTREAM=1",
+    ],
+    includes = [
+        "include",
+    ],
+    visibility = ["//visibility:public"],
+)"""
+
+    native.new_http_archive(
+        name = "tclap_tar",
+        url = "https://storage.googleapis.com/istio-build-deps/tclap-1.2.1.tar.gz",
+        strip_prefix = "tclap-1.2.1",
+        build_file_content = BUILD,
+    )
+    if bind:
+        native.bind(
+            name = "tclap",
+            actual = "@tclap_tar//:tclap",
+        )
+
+def lightstep_repositories(bind=True):
+    BUILD = """
+load("@protobuf_git//:protobuf.bzl", "cc_proto_library")
+
+genrule(
+    name = "envoy_carrier_pb",
+    srcs = ["src/c++11/envoy/envoy_carrier.proto"],
+    outs = ["lightstep/envoy_carrier.proto"],
+    cmd = "cp $(SRCS) $@",
+)
+
+cc_proto_library(
+    name = "envoy_carrier_proto",
+    srcs = ["lightstep/envoy_carrier.proto"],
+    include = ".",
+    deps = [
+        "//external:cc_wkt_protos",
+    ],
+    protoc = "//external:protoc",
+    default_runtime = "//external:protobuf",
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "lightstep_core",
+    srcs = [
+        "src/c++11/impl.cc",
+        "src/c++11/span.cc",
+        "src/c++11/tracer.cc",
+        "src/c++11/util.cc",
+    ],
+    hdrs = [
+        "src/c++11/lightstep/impl.h",
+        "src/c++11/lightstep/options.h",
+        "src/c++11/lightstep/propagation.h",
+        "src/c++11/lightstep/envoy.h",
```

Comment:
Buildifier: `src/c++11/lightstep/envoy.h` should be listed first (alphabetical order).